src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/TensorsInfo.cs

```diff
@@ -262,11 +262,127 @@ public TensorsData GetTensorsData()
             ret = Interop.Util.CreateTensorsData(_handle, out tensorsData_h);
             NNStreamer.CheckException(ret, "unable to create the tensorsData object");
 
-            retTensorData = TensorsData.CreateFromNativeHandle(tensorsData_h);
+            retTensorData = TensorsData.CreateFromNativeHandle(tensorsData_h, _handle, false);
 
             return retTensorData;
         }
 
+        /// <summary>
+        /// Gets the tensor name with given index.
+        /// </summary>
+        /// <param name="idx">The index of the tensor.</param>
+        /// <returns>The tensor name</returns>
+        /// <feature>http://tizen.org/feature/machine_learning.inference</feature>
+        /// <exception cref="ArgumentException">Thrown when the method failed due to an invalid parameter.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public string GetTensorName(int idx)
+        {
+            NNStreamer.CheckNNStreamerSupport();
+
+            CheckIndexBoundary(idx);
+            return _infoList[idx].Name;
+        }
+
+        /// <summary>
+        /// Gets the the hash code of this TensorsInfo object
+        /// </summary>
+        /// <returns>The hash code</returns>
+        /// <since_tizen> 7 </since_tizen>
+        public override int GetHashCode()
+        {
+            return this._infoList.GetHashCode();
```

Comment:
Equal object should be return same hashcode

Reply:
A way to implement GetHashCode with Collection
``` c#
public override int GetHashCode()
{
    unchecked
    {
        int hash = 19;
        foreach (var foo in foos)
        {
            hash = hash * 31 + foo.GetHashCode();
        }
        return hash;
    }
}
```

Reply:
ACK!